@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -756,72 +750,72 @@ export class TiltViewClipPlanesEvaluator extends TopViewClipPlanesEvaluator {
         const viewRanges = { ...this.minimumViewRange };
 
         // Near plane calculus is pretty straightforward and does not depend on camera tilt:
-        const cameraAltitude = this.getCameraAltitude(camera, projection);
-        viewRanges.near = cameraAltitude - this.maxElevation;
+        viewRanges.near = projection.groundDistance(camera.position) - this.maxElevation;
 
         // Far plane calculation requires different approaches depending from camera projection:
         // - perspective
         // - orthographic
-        const cameraToOrigin = this.m_tmpVectors[0].copy(camera.position).negate();
         const r = EarthConstants.EQUATORIAL_RADIUS;
-        const d = cameraToOrigin.length();
-        let farPlane: number;
+        const maxR = r + this.maxElevation;
+        const d = camera.position.length();
+
         if (camera instanceof THREE.PerspectiveCamera) {
             const halfVerticalFovAngle = THREE.MathUtils.degToRad(camera.fov / 2);
 
-            // Ratio of the depth of the camera compared to the distance between the center of the
-            // screen and the middle of the top of the screen (here set to be value 1). We just need
-            // this to be a ratio because we are interested in just computing the direction of the
-            // upper right corner and not any specific length.
-            const zLength = 1 / Math.tan(halfVerticalFovAngle);
-
-            // Camera space direction vector along the top right frustum edge (of the camera).
-            const upperRightDirection = this.m_tmpVectors[2];
-            upperRightDirection.set(camera.aspect, 1, -zLength);
-
             // Now we need to account for camera tilt and frustum volume, so the longest
             // frustum edge does not intersects with sphere, it takes the worst case
             // scenario regardless of camera tilt, so may be improved little bit with more
             // sophisticated algorithm.
             viewRanges.near *= Math.cos(halfVerticalFovAngle);
-            // Step-wise calculate angle between camera eye vector and tangent
-
-            // Calculate angle between surface normal(below camera position) and tangent.
-            const alpha = Math.asin(r / d);
 
-            // Calculate angle between look at and surface normal(below camera position)
-            const cameraPitch = this.getCameraPitch(cameraToOrigin, camera);
+            // Ratio of the depth of the camera compared to the distance between the center of the
+            // screen and the middle of the top of the screen (here set to be value 1). We just need
+            // this to be a ratio because we are interested in just computing the direction of the
+            // upper right corner and not any specific length.
+            const zLength = 1 / Math.tan(halfVerticalFovAngle);
```

Comment:
it would be good to move `zLength`, `upperRightDirection` and the setting of the ray into a separate function, because the `upperRightDirection` is only valid until line 787, i.e. when fovSphereIntersection is initialised, because that takes over ownership of `this.m_tmpVectors[0]`, and my putting it in a method, you are limiting the leak of these variables being used later for example if the code is refactored.

Reply:
All this is changed in follow-up PRs, zLength is not there anymore and I've introduced a separate function to compute ray intersections, cause it's needed in more places.

Reply:
Ok, cool.